main.cpp

```diff
@@ -172,8 +190,20 @@ bool TestWriteReadSimple()
         printf("\nERROR: Device not ready, tests failed\n");
         return false;
     }
+
+    //Send WREN
+    if (!WriteEnable()) {
+        return false;
+    }
+
+    if( !WaitForMemReady()) {
+        printf("\nERROR: Device not ready, tests failed\n");
+        return false;
+    }
     
-    result = myQspi->write(0x02, 0, flash_addr, tx_buf, &buf_len );
+    printf(">>>>>>>>>>>>>>>>>>>>START WRITING...\n");
+    /* bit mask last 8 bits of the adress */
+    result = myQspi->write(QSPI_PAGE_PROG_CMD, -1, 0, (flash_addr & 0x00FFFF00), tx_buf, &buf_len );
```

Comment:
why is the bit masking in write here?

Reply:
The chapter Page Program in MX25R6435F document says:
> The last address byte (the 8 least significant address bits, A7-A0) should be set to 0 for 256 bytes page program. If A7-A0 are not all zero, transmitted data that exceed
> page length are programmed from the starting address (24-bit address that last 8 bit are all 0) of currently selected page.

This cannot be done inside the driver, because it depends on the page size of this specific memory.